src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestUpdate.java

```diff
@@ -924,10 +931,10 @@ public void testGetSubmodules() throws Exception {
         assertFixSubmoduleUrlsThrows();
     }
 
-    @Ignore("TODO see comment in CliGitAPIImplTest.setupGitAPI")
     @Test
     public void testSubmoduleUpdateShallow() throws Exception {
         WorkingArea remote = setupRepositoryWithSubmodule();
+        w.cgit().allowFileProtocol(); // CLI git 2.38.1 requires protocol.file.allow=always for this test
```

Comment:
I think the comment can be dropped here—Javadoc of the method being called explains this in detail.

Reply:
Done in 7f8b9879477b0fd68b667720e9cbf46237e20dbe